clients/drcachesim/tests/burst_syscall_inject.cpp

```diff
@@ -87,6 +90,25 @@ do_some_syscalls()
 {
     getpid();
     gettid();
+
+    // Make some failing sigaction syscalls, which we record such that
+    // syscall_mix will count the failure codes.
+    // We bypass libc to ensure these make it to the syscall itself.
```

Comment:
Oh you mean libc does some of these checks itself to avoid the overhead of an OS syscall when it's only going to end in failure?

Reply:
Yes, IIRC the invalid signal number never made it to the syscall; and the fault one had some other issue: it actually crashed I think inside libc.  Libc has a different definition of the mask so it has to translate to another copy.